apps/framework-cli-e2e/test/templates.test.ts

```diff
@@ -443,24 +472,32 @@ describe("Moose Templates", () => {
 
     it("should successfully ingest data and verify through consumption API", async function () {
       const eventId = randomUUID();
-      const response = await fetch(`${TEST_CONFIG.server.url}/ingest/Foo`, {
-        method: "POST",
-        headers: { "Content-Type": "application/json" },
-        body: JSON.stringify({
-          primaryKey: eventId,
-          timestamp: TEST_CONFIG.timestamp,
-          optionalText: "Hello world",
-        }),
-      });
 
-      if (!response.ok) {
-        console.error("Response code:", response.status);
-        const text = await response.text();
-        console.error(`Test request failed: ${text}`);
-        throw new Error(`${response.status}: ${text}`);
+      // Send multiple records to trigger batch write (batch size is likely 1000+)
+      const recordsToSend = 50; // Send enough to trigger a batch
+      const responses = [];
+
+      for (let i = 0; i < recordsToSend; i++) {
+        const response = await fetch(`${TEST_CONFIG.server.url}/ingest/Foo`, {
+          method: "POST",
+          headers: { "Content-Type": "application/json" },
+          body: JSON.stringify({
+            primaryKey: i === 0 ? eventId : randomUUID(), // Use eventId for first record for verification
+            timestamp: TEST_CONFIG.timestamp,
+            optionalText: `Hello world ${i}`,
+          }),
+        });
+
+        if (!response.ok) {
+          console.error("Response code:", response.status);
+          const text = await response.text();
+          console.error(`Test request failed: ${text}`);
+          throw new Error(`${response.status}: ${text}`);
+        }
+        responses.push(response);
       }
 
-      await utils.waitForDBWrite(devProcess!, "Bar", 1);
+      await utils.waitForDBWrite(devProcess!, "Bar", recordsToSend);
```

Comment:
what if the test script is not sending fast enough and the framework does that in two batches?

Reply:
Good call - I will change that

Reply:
I actually changed the waitForDbWrite to accumulate all the count it sees. That way we can actually count the total number of records landed in the db